### ISSUE TITLE
Bad .wav duration

### ISSUE THREAD

**Issue by Simanas** [librosa/librosa#749]
Hello! 

First of all thank you guys for developing this amazing tool! 
I use it a lot and I come across one case where librosa somehow fails to determine correct file duration therefore fails to load it. I have a .wav file that is  stereo track in 44100 kHZ, 32-bit float. 
In Audacity this file has 3 mins and 6.87 seconds length. When I do "librosa.get_duration(..." I get: 189.567

So that is off from original length by ~ 3 seconds. Now when I do "librosa.load(..." on that particular file I get an error: 

```
File "/home/simanas/.virtualenvs/PhonicMind/local/lib/python2.7/site-packages/librosa/core/audio.py", line 150, in load
    y = y.reshape((-1, n_channels)).T
ValueError: cannot reshape array of size 16394065 into shape (2)
```

But if I add duration=187 to librosa.load function then it all works good.

Here is the file that is giving this error: https://files.fm/u/j44v5jgc

Appreciate your help!

**Comment by stefan-balke** [comment]
Hey,

although this not explain why it doesn't work, we nowadays recommend using `pysoundfile` for IO handling.
Could you please try it with this library before we investigate further?

Thanks!

**Comment by Simanas** [comment]
Hmmm... ok so is there some way I can simply plug pysoundfile in to librosa and continue using standard librosa syntax or do I have to do my own boilerplate code to achieve the same type of interface like librosa.load has, where I can enter offset and duration in seconds without worrying about sample rates, etc...?

**Comment by stefan-balke** [comment]
Well, when you look at:

http://pysoundfile.readthedocs.io/en/0.9.0/#soundfile.read

That's what they offer. So, for your, `start`, `stop`, and `frames` will be important.

But you could also just copy `librosa.load()` and exchange the `audioread` call.
https://github.com/librosa/librosa/blob/master/librosa/core/audio.py#L112

Not tested, but should work with minor adjustments...

**Comment by Simanas** [comment]
Well yes that might work... but that is basically trashing everything that librosa has to offer on files reading and changing it with more complicated solution that is not really giving any gains to my code base, only adds unnecessary complexity. 

One of the most useful librosa feature that I use a lot is `librosa.get_duration`, so that would have to be addressed with it somehow as well which would add even more complexity... 

In short - I use librosa because it's so amazingly simple to use and gives me everything I need without getting to deep in to various low level audio loading processing and saving things. This is why I love it, this is why we all loving it!

So I would rather vote for figuring out why librosa fails to determine .wav file duration correctly than switching to `pysoundfile`. :)

**Comment by stefan-balke** [comment]
Yes, on the long run those wrapping functions will be replaced. But, as you see, the solution only works for 95% of the WAV files...as long as you don't have one of those files you're fine :)
Obviously, yours is different. The thing is, that those errors are hardly related to `librosa` but to one of the backends you are using. So, everything we discover here will have to be forwarded to `ffmpeg` or whatever you are using anyway...just warning you! `pysoundfile` might be the lesser "evil" then...

**Comment by stefan-balke** [comment]
Ah, also please use this Issue template: https://raw.githubusercontent.com/librosa/librosa/master/ISSUE_TEMPLATE.md

**Comment by bmcfee** [comment]
@Simanas without example code showing how you're loading the data or calculating duration, it's difficult to diagnose the error.

There's a known issue with `get_duration` on filename inputs, where if the file header is corrupted, the result will be incorrect.  I suspect that's what's going on here, and also what's causing the load to fail.  I bet that if you re-encode the wav file (say, by loading into audacity and saving as a new file), this problem should go away.

`audioread` is not the most robust of toolkits when it comes to badly formed files, and as @stefan-balke mentions, pysoundfile is a much better option in a lot of ways.  As I've said elsewhere, we're likely to deprecate IO functionality in the future (once libsndfile gets mp3 support) in favor of using pysoundfile, though we might include shims to make the transition easier.
